### PR TITLE
install: store secrets in an env file for ibmcloud-powervs

### PR DESCRIPTION
**Pull request by Amulyam24**
This PR adds changes to store the api key and related credentials in an env file which is set in kustomization.

Fixes: part of #1136

### CODE REVIEW

**Comment by jtumber-ibm** [comment]
Thanks for this change @Amulyam24 please hold off merging it for now as it is a breaking change for our end-2-end testing framework.

File: install/overlays/ibmcloud/kustomization.yaml [hunk 1]
```diff
@@ -43,13 +43,14 @@ secretGenerator:
...
-  - IBMCLOUD_IAM_ENDPOINT="" #set
-  - IBMCLOUD_ZONE="" #set
+  # the below file should be in this format
+  # $ cat cloud-credentials.env
+  # IBMCLOUD_IAM_PROFILE_ID=...
+  # IBMCLOUD_API_KEY=...
+  # IBMCLOUD_IAM_ENDPOINT=...
+  # IBMCLOUD_ZONE=...
+  envs:
+  - cloud-credentials.env
```
Comment (stevenhorsman) [inline install/overlays/ibmcloud/kustomization.yaml:53]: Can you separate out the ibmcloud kustomization change from this PR please as this update will break the CI, so it needs to go in together with those changes

Comment (stevenhorsman) [reply install/overlays/ibmcloud/kustomization.yaml:53]: Sorry, I missed that JT had already commented about this

Comment (Amulyam24) [reply install/overlays/ibmcloud/kustomization.yaml]: Hi @stevenhorsman, no problem! I have separated out the kustomization changes for ibmcloud. PTAL

Comment (Amulyam24) [reply install/overlays/ibmcloud/kustomization.yaml:53]: Sorry, I think I got it wrong. I don't mind holding off merging this PR until the e2e breakage is sorted out. Re-added the ibmcloud changes.

**Comment by Amulyam24** [comment]
Hi @jtumber-ibm @stevenhorsman, I rebased this PR. Is this still on `hold`?

**Comment by jtumber-ibm** [comment]
> Hi @jtumber-ibm @stevenhorsman, I rebased this PR. Is this still on `hold`?

Hi @Amulyam24, thanks for keeping track of this. Unfortunately we don't have the dev time to update our internal e2e testing for this change. So if you'd like to remove the ibmcloud vpc changes and progress just with the powervs we will pick up this change when we have some more time.

**Review by jtumber-ibm — APPROVED** [review]
lgtm, thanks

**Comment by Amulyam24** [comment]
> > Hi @jtumber-ibm @stevenhorsman, I rebased this PR. Is this still on `hold`?
> 
> Hi @Amulyam24, thanks for keeping track of this. Unfortunately we don't have the dev time to update our internal e2e testing for this change. So if you'd like to remove the ibmcloud vpc changes and progress just with the powervs we will pick up this change when we have some more time.

Thanks @jtumber-ibm. I made the changes, PTAL.

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks

**Comment by Amulyam24** [comment]
Hi @stevenhorsman @jtumber-ibm, rebased the PR. can it be merged?

**Pull request merged**